lib/inertia_rails/meta_tag.rb

```diff
@@ -0,0 +1,100 @@
+module InertiaRails
+  class MetaTag
+    # Copied from Inertia.js
+    UNARY_TAGS = %i[
+      area base br col embed hr img input keygen link meta param source track wbr
+    ].freeze
+
+    def initialize(tag_name: nil, head_key: nil, raw: false, allow_duplicates: false, **tag_data)
+      @is_shortened_title_tag = shortened_title_tag?(tag_name, tag_data)
+      @tag_name = determine_tag_name(tag_name)
+      @allow_duplicates = allow_duplicates
+      @raw = raw
+      @tag_data = build_tag_data(tag_data)
+      @head_key = head_key || generate_head_key
+    end
+
+    def as_json(**options)
+      {
+        :tagName => @tag_name,
+        :headKey => @head_key,
+        **@tag_data.transform_keys { |k| k.to_s.camelize(:lower).to_sym }
+      }
+    end
+
+    def to_tag(tag_helper)
+      data = @tag_data.deep_dup
+        .merge({ inertia: @head_key })
+
+      inner_content = case @tag_name
+                      when *UNARY_TAGS
+                        nil
+                      when :script
+                        handle_script_content(data.delete(:inner_content))
+                      else
+                        data.delete(:inner_content)
+                      end
+
+      tag_helper.public_send(@tag_name, *[inner_content].compact, **data.transform_keys { |k| k.to_s.tr('_','-').to_sym })
+    end
+
+    def [](key)
+      return @tag_name if key == :tag_name
+      return @head_key if key == :head_key
+      @tag_data[key.to_sym]
+    end
+
+    private
+
+    def generate_head_key
+      generate_meta_head_key || "#{@tag_name}-#{tag_digest}"
+    end
+
+    def handle_script_content(content)
+      case content
+      when String
+        @raw ? content.html_safe : ERB::Util.html_escape(content)
```

Comment:
There's an (undocumented) escape hatch in case you want to render eval'able `<script>` tags. It's undocumented because I can't think of a good reason why you'd actually want to render a script tag like this.

Reply:
Yea I dug back into my thinking and am going to make a couple changes. I don't think we need eval'able scripts after reconsidering it (I was originally thinking about analytics tracking scripts, but that doesn't really belong in this feature). And I need to escape the content on Inertia requests... the browser normally ignores script content created by `dangerouslySetInnerHTML`, but Inertia.js's `<Head>` renderer overrides it and make it eval'able again.